player/lib/player/impl.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Player.Impl do
+
+  @player_speed 1
+  defp move_player(player = %{moving: false}) do
+    player
+  end
```

Comment:
There seem to be at least three versions of move up/down/left/right on tick. Could this be extracted out into a common library? Protocols might help (although you might not need them...)